File: foxogram-common/src/main/java/su/foxogram/dto/api/response/MessagesDTO.java
package su.foxogram.dto.api.response;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Getter;
import lombok.Setter;
import su.foxogram.model.Message;

import java.util.List;

@Getter
@Setter
@Schema(name = "Messages")
public class MessagesDTO {

	private List<MessageDTO> messages;

	public MessagesDTO(List<Message> messages) {
		for (Message message : messages) {
			this.messages.add(new MessageDTO(message, true));
		}
	}
}

File: src/main/java/su/foxogram/dto/api/response/MessagesDTO.kt
package su.foxogram.dto.api.response

import io.swagger.v3.oas.annotations.media.Schema
import lombok.Getter
import lombok.Setter
import su.foxogram.model.Message

@Getter
@Setter
@Schema(name = "Messages")
data class MessagesDTO(val messages: MutableList<Message>) {
    var listMessages: MutableList<MessageDTO?>? = null

    init {
        for (message in messages) {
            this.listMessages!!.add(MessageDTO(message, true))
        }
    }
}
